src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -178,10 +178,16 @@ internal HttpClient GetHttpClient()
                         switch (_authenticationScheme)
                         {
                             case AuthenticationSchemes.Basic:
-                                if (credentials.UserName.UserName != string.Empty)
+                                if (credentials.UserName.UserName == null)
                                 {
-                                    creds = new NetworkCredential(credentials.UserName.UserName, credentials.UserName.Password);
+                                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("userName");
                                 }
+                                if (credentials.UserName.UserName == string.Empty)
```

Comment:
Nope. This is the same behavior as the desktop. We would like to be consistent with desktop.